awx/main/db/profiled_pg/base.py

```diff
@@ -0,0 +1,101 @@
+import os
+import traceback
+import uuid
+
+from django.core.cache import cache
+from django.core.cache.backends.locmem import LocMemCache
+from django.db.backends.postgresql.base import DatabaseWrapper
+
+from awx.main.utils import memoize
+
+__loc__ = LocMemCache(str(uuid.uuid4()), {})
+__all__ = ['DatabaseWrapper']
+
+
+class RecordedQueryLog(object):
+
+    dest = '/var/lib/awx/profile'
+
+    def __init__(self, log, db):
+        self.log = log
+        self.db = db
+        self.threshold = cache.get('awx-profile-sql-threshold')
+
+    def append(self, query):
```

Comment:
I'm relying on Django to call `queries_log.append(...)` per-query when `force_debug_cursor` is True.  This is long-lived code in Django and _very_ slow-moving (so hopefully not subject to major changes).

https://github.com/django/django/blob/978ad6d4c043f5004005b673c6e7bb8f415a35dc/django/db/backends/utils.py#L96-L111